### PR TITLE
Fix ITT unit-tests if PyTorch is compiled with `USE_ITT=OFF`

### PR DESCRIPTION
**Pull request by jingxu10**
Fixes https://github.com/pytorch/pytorch/pull/84848#discussion_r986329680

@malfet @slgong-fb

### CODE REVIEW

**Comment by jingxu10** [comment]
Hi @malfet , @slgong-fb , could you take a review of this PR? This should fix the internal blocking issue.

File: test/test_autograd.py [hunk 1]
```diff
@@ -8087,6 +8087,8 @@ def test_rnn_backward_to_input_but_not_parameters(self, device):
         out.sum().backward()
         self.assertFalse(s.grad is None or s.grad.abs().sum().item() == 0)
 
+    @unittest.skipIf(IS_ARM64, "ITT is not available on ARM")
```
Comment (slgong-fb) [inline test/test_autograd.py:8090]: If we have `is_available()`, isnt `IS_ARM64` a redundant check?

File: torch/profiler/itt.py [hunk 2]
```diff
@@ -8,14 +8,26 @@ class _ITTStub(object):
         def _fail(*args, **kwargs):
             raise RuntimeError("ITT functions not installed. Are you sure you have a ITT build?")
 
+        @staticmethod
+        def _is_available(*args, **kwargs):
+            return False
+
+        is_available = _is_available
```
Comment (malfet) [inline torch/profiler/itt.py:15]: Nit
```suggestion
        @staticmethod
        def is_available():
            return False
```

**Review by malfet — APPROVED** [review]
Looks good to me, but please address @slgong-fb comment, that IS_ARM64 check is no longer needed if we have an availability mechanism in place.

Also, do not overcomplicate the wrapper, just default is_available as a static method that returns False

File: test/test_autograd.py (hunk 1 above)
Comment (jingxu10) [reply test/test_autograd.py:8090]: Yeah, I think so. Let's use `is_available()` then. I'll remove the `IS_ARM64` check.

File: torch/csrc/itt_wrapper.cpp [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 #include <ittnotify.h>
+#include <torch/csrc/profiler/api.h>
 
 namespace torch {
 namespace profiler {
 __itt_domain* _itt_domain = __itt_domain_create("PyTorch");
 
+TORCH_API bool itt_is_available() {
+  return torch::profiler::impl::ittStubs()->enabled();
+}
```
Comment (malfet) [inline torch/csrc/itt_wrapper.cpp:11]: Just curious, when this function could return false?

File: test/test_autograd.py [hunk 4]
```diff
@@ -8087,6 +8087,8 @@ def test_rnn_backward_to_input_but_not_parameters(self, device):
...
+    @unittest.skipIf(not torch.profiler.itt.is_available(), "ITT is required")
     @onlyCPU
```
Comment (jingxu10) [inline test/test_autograd.py:8092]: @slgong-fb @malfet Shall I remove this `@onlyCPU` check as well? Only check with `is_available()`?

Comment (jingxu10) [reply test/test_autograd.py:8092]: Hi @slgong-fb @malfet , any comments here?

Comment (slgong-fb) [reply test/test_autograd.py:8092]: I am not sure 100%, but we do not need `onlyCPU`. ITT's scope is covering both CPU and GPU, right? So, as long as ITT is there, we can run this test?

File: test/profiler/test_profiler.py [hunk 5]
```diff
@@ -129,6 +129,7 @@ def custom_layer(input_ten):
             q.backward()
 
 @unittest.skipIf(IS_ARM64, "ITT is not available on ARM")
```
Comment (slgong-fb) [inline test/profiler/test_profiler.py:131]: same here. we may not need `IS_ARM64`

File: test/test_autograd.py (hunk 4 above)
Comment (jingxu10) [reply test/test_autograd.py:8092]: Yeah, I agree, as long as ITT is there, we can run this test.

**Comment by jingxu10** [comment]
@slgong-fb merged

**Pull request closed without merging**